triggers/service/src/main/scala/com/digitalasset/daml/lf/engine/trigger/Server.scala

```diff
@@ -268,40 +270,43 @@ class Server(
             }
           }
         }
-        Directive { inner =>
-          auth {
-            // Authorization successful - pass token to continuation
-            case Some(authorization) => inner(Tuple1(Some(authorization)))
-            // Authorization failed - login and retry on callback request.
-            case None => { ctx =>
-              val requestId = UUID.randomUUID()
-              authCallbacks.update(
-                requestId, {
-                  auth {
-                    case None => {
-                      // Authorization failed after login - respond with 401
-                      // TODO[AH] Add WWW-Authenticate header
-                      complete(errorResponse(StatusCodes.Unauthorized))
-                    }
-                    case Some(authorization) =>
-                      // Authorization successful after login - use old request context and pass token to continuation.
-                      mapRequestContext(_ => ctx) {
-                        inner(Tuple1(Some(authorization)))
+        auth.flatMap {
+          // Authorization successful - pass token to continuation
+          case Some(authorization) => provide(Some(authorization))
+          // Authorization failed - login and retry on callback request.
+          case None =>
+            // Ensure that the request is fully uploaded.
+            toStrictEntity(httpEntityUploadTimeout, maxHttpEntityUploadSize.toLong).tflatMap { _ =>
```

Comment:
Maybe we should just store it as a long given that we only ever use it as that?

Reply:
Good point, fixed.